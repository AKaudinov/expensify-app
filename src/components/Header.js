import React from 'react'
import {NavLink} from "react-router-dom";
import {connect } from 'react-redux';
import {startLogOut} from "../actions/auth";

export const Header = ({logout}) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" exact={true} activeClassName="active-link">Home</NavLink>
        <NavLink to="/create" activeClassName="active-link">Create Expense</NavLink>
        {/*<NavLink to="/edit" activeClassName="active-link">Edit Expense</NavLink>*/}
        {/*<NavLink to="/help" activeClassName="active-link">Help</NavLink>*/}
        <button onClick={logout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => {
  return {
      logout: () => {dispatch(startLogOut())}
  }
};


export default connect(undefined, mapDispatchToProps)(Header);
