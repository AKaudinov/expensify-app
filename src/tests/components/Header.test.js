//snapshot testing - track changes over time

// react-test-renderer - render components inside JS code and assert the component
import ReactShallowRenderer from 'react-test-renderer/shallow';

//shallow rendering < no interaction, and also with no child components
//full DOM rendereing < allows interaction and renders all child components

import React from 'react';
import Header from '../../components/Header';
import {shallow} from 'enzyme' //import shallow renderer from enzyme

test('Should render Header correctly', () => {
   // const renderer = new ReactShallowRenderer();
   // renderer.render(<Header/>); //renderer renders the jsx you specify, in this case header jsx
   // expect(renderer.getRenderOutput()).toMatchSnapshot(); //first time the test runs, it will pass, and jest will create
   //  //a snapshot of jsx, and 2nd time it will compare the rendering to the created snapshot


   //Enzyme testing
   const wrapper = shallow(<Header />);
   expect(wrapper).toMatchSnapshot(); //make enzyme work with the snapshots via enzyme-to-json library
});


//enzyme released by airbnb for react js and testing it
