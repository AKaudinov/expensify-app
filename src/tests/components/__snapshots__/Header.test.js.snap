// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="active-link"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="active-link"
    ariaCurrent="true"
    to="/create"
  >
    Create Expense
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
